pygen/pygen_src/riscv_instr_stream.py

```diff
@@ -215,9 +215,15 @@ def randomize_instr(self, instr, is_in_debug = 0):
             exclude_instr.append(riscv_instr_name_t.C_ADDI16SP.name)
             exclude_instr.append(riscv_instr_name_t.C_LWSP.name)
             exclude_instr.append(riscv_instr_name_t.C_LDSP.name)
-        if is_in_debug and (not cfg.enable_ebreak_in_debug_rom):
-            exclude_instr.append(riscv_instr_name_t.EBREAK.name)
-            exclude_instr.append(riscv_instr_name_t.C_EBREAK.name)
+        # Post-process the allowed_instr and exclude_instr lists to handle
+        # adding ebreak instructions into the debug ROM.
+        if is_in_debug:
+            if (cfg.no_ebreak and cfg.enable_ebreak_in_debug_rom):
+                allowed_instr.append(riscv_instr_name_t.EBREAK.name)
+                allowed_instr.append(riscv_instr_name_t.C_EBREAK.name)

```

Comment:
Sounds good to me, just a minor comment: Instead of `appending` once for each item, better to call `extend` and add all items together:
```suggestion
                allowed_instr.extend([riscv_instr_name_t.EBREAK.name,
                                    riscv_instr_name_t.C_EBREAK.name])
```

Reply:
good point, will change this